### DISCUSSION TITLE
Add Support for Multiple Roles

### DISCUSSION THREAD

**Discussion by josh-fisher** [maxcountryman/axum-login#88] [Ideas]
Hi. Hope all is well and thanks for a great crate.

Was wondering if there were plans to support multiple Roles and if so if you had a roadmap for implementation. As far as I can tell the current codebase can only support testing 1 user role.

If I'm mistaken, does anyone have an example of how to do so?

**Comment by maxcountryman** [comment]
Just to clarify, are you asking for the ability to accept multiple roles during route validation? The current implementation uses `Range` for this.

**Comment by czocher** [comment]
FYI if you require a user to have multiple roles, usually that's a use case for permission-based authentication, not role-based

**Comment by maxcountryman** [comment]
If you have a look at the new design, I'd be curious to know your thoughts regarding this. Does it satisfy the use case you had in mind?
